hs_file_types/models/geofeature.py

```diff
@@ -249,125 +249,69 @@ def check_files_for_aggregation_type(cls, files):
-                    # user selected a folder to create aggregation
-                    # make all the files in the selected folder as part of the aggregation
-                    logical_file.add_resource_files_in_folder(resource, folder_path)
-
-                log.info("GeoFeature aggregation - files were added to the aggregation.")
-                add_metadata(resource, meta_dict, xml_file, logical_file)
-                log.info("GeoFeature aggregation and resource level metadata updated.")
-                reset_title = logical_file.dataset_name == base_file_name
-                logical_file._finalize(user, resource, folder_created=aggregation_folder_created,
-                                       res_files_to_delete=res_files_to_delete,
-                                       reset_title=reset_title)
-
-                file_type_success = True
-                post_add_geofeature_aggregation.send(
-                    sender=AbstractLogicalFile,
-                    resource=resource,
-                    file=logical_file
-                )
-            except Exception as ex:
-                msg = "GeoFeature aggregation type. Error when creating aggregation type. Error:{}"
-                msg = msg.format(ex.message)
-                log.exception(msg)
-            finally:
-                # remove temp dir
-                if os.path.isdir(temp_dir):
-                    shutil.rmtree(temp_dir)
-
-        if not file_type_success:
-            aggregation_from_folder = folder_path is not None
-            cls._cleanup_on_fail_to_create_aggregation(user, resource, upload_folder,
-                                                       file_folder, aggregation_from_folder)
-            raise ValidationError(msg)
+                        files_to_upload = []
+                        res_files_for_aggr = shp_res_files
+
+                    # create a GeoFeature logical file object
+                    logical_file = cls.create_aggregation(dataset_name=base_file_name,
+                                                          resource=resource,
+                                                          res_files=res_files_for_aggr,
+                                                          new_files_to_upload=files_to_upload,
+                                                          folder_path=upload_folder)
+
+                    log.info("GeoFeature aggregation - files were added to the aggregation.")
+                    add_metadata(resource, meta_dict, xml_file, logical_file)
+                    log.info("GeoFeature aggregation and resource level metadata updated.")
+
+                    file_type_success = True
+                    ft_ctx.logical_file = logical_file
+                    ft_ctx.res_files_to_delete = res_files_to_delete
+                except Exception as ex:
+                    msg = "GeoFeature aggregation type. Error when creating aggregation " \
+                          "type. Error:{}"
```

Comment:
msg is already defined on line 282, so line 308 and 309 can be removed.

Reply:
good catch. will fix it.

Reply:
fixed.